pypeit/wavecalib.py

```diff
@@ -741,6 +764,97 @@ def build_wv_calib(self, arccen, method, skip_QA=False,
+            order_dets (np.ndarray): 
+                Orders on the each detector
+
+        Returns:
+            bool: True if any of the echelle orders were 
+            successfully redone
+        """
+
+        # Make this outside the for loop..
+        #bad_orders = self.slits.ech_order[np.where(bad_rms)[0]]
+        fixed = False
+
+        for idet in range(len(dets)):
+            in_det = np.in1d(bad_orders, order_dets[idet])
+            if not np.any(in_det):
+                continue
+            # Are there few enough?
+            # TODO -- make max_bad a parameter
+            max_bad = len(order_dets[idet])//10 + 1
+            if np.sum(in_det) > max_bad:
+                msgs.warn(f"Too many bad orders in detector={dets[idet]} to attempt a refit.")
+                continue
+            # Loop
+            for order in bad_orders[in_det]:
+                iord = np.where(self.slits.ech_order == order)[0][0]
+                # Predict the wavelengths
+                nspec = self.arccen.shape[0]
+                spec_vec_norm = np.linspace(0,1,nspec)
+                wv_order_mod = self.wv_calib.wv_fit2d[idet].eval(spec_vec_norm, 
+                                    x2=np.ones_like(spec_vec_norm)*order)/order
+
+                # Link me
+                tcent, spec_cont_sub, patt_dict_slit, tot_llist = autoid.match_to_arxiv(
+                    self.lamps, self.arccen[:,iord], wv_order_mod, 
+                    self.arcspec_arxiv[:, iord],  self.wave_soln_arxiv[:,iord],
+                    self.par['nreid_min'], 
+                match_toler=self.par['match_toler'], 
+                nonlinear_counts=self.nonlinear_counts, 
+                sigdetect=wvutils.parse_param(self.par, 'sigdetect', iord),
+                fwhm=self.par['fwhm'])
+
+                if not patt_dict_slit['acceptable']:
+                    msgs.warn(f"Order {order} is still not acceptable after attempt to reidentify.")
+                    continue
+
+                # Fit me -- RMS may be too high again
+                n_final = wvutils.parse_param(self.par, 'n_final', iord)
+                # TODO - Make this cheaper
+                final_fit = wv_fitting.fit_slit(
+                    spec_cont_sub, patt_dict_slit, tcent, tot_llist, 
+                    match_toler=self.par['match_toler'], 
+                    func=self.par['func'], 
+                    n_first=self.par['n_first'],
+                    #n_first=3,
+                    sigrej_first=self.par['sigrej_first'], 
+                    n_final=n_final, 
+                    sigrej_final=2.)
+                msgs.info(f"New RMS for redo of order={order}: {final_fit['rms']}")
+
+                # Keep?
+                # TODO -- Make this a parameter?
+                increase_rms = 1.5
+                if final_fit['rms'] < increase_rms*self.par['rms_threshold']* np.median(self.measured_fwhms)/self.par['fwhm']:
+                    # TODO -- This is repeated from lines 718-725
```

Comment:
I don't think these line numbers are correct... maybe better to give the place the function is called?

Reply:
done